diesel_derives/src/queryable.rs

```diff
@@ -49,3 +63,100 @@ pub fn derive(item: syn::DeriveInput) -> Result<quote::Tokens, Diagnostic> {
         },
     ))
 }
+
+fn derive_checked(
+    item: syn::DeriveInput,
+    check_types: MetaItem,
+) -> Result<quote::Tokens, Diagnostic> {
+    let model = Model::from_item(&item)?;
+
+    let struct_name = item.ident;
+    let backend = check_types.nested_item("backend")?.ty_value()?;
+    let sql_tys = model.fields().iter().map(|f| sql_type(f, &model));
+    let field_exprs = model
+        .fields()
+        .iter()
+        .enumerate()
+        .map(|(i, f)| field_expr(i, f, &model, &backend));
+
+    let (impl_generics, ty_generics, where_clause) = item.generics.split_for_impl();
+
+    Ok(wrap_in_dummy_mod(
+        model.dummy_mod_name("queryable"),
+        quote! {
+            use self::diesel::deserialize::{self, Queryable, FromSqlRow};
+
+            pub enum DummyRow {}
+
+            impl<ST> FromSqlRow<ST, #backend> for DummyRow {
+                fn build_from_row<R: self::diesel::row::Row<#backend>>(_: &mut R)
+                    -> deserialize::Result<Self>
+                {
+                    use self::std::result::Result::Err;
+                    use self::std::convert::Into;
+
+                    Err("`#[check_types]` is only for debugging purposes".into())
```

Comment:
I think that should rather panic than return a `Err`, because this is clearly a programmer mistake. 
(We could even think about using `compile_error!` if this typechecks correctly.)

Reply:
Nah, because you could have lots of code, add a column to you `Queryable` struct and still want the project to compile instead of throwing 20 000 compile errors telling you that `#[check_types]` is for debugging only.